substrate/client/transaction-pool/tests/zombienet/mod.rs

```diff
@@ -0,0 +1,103 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Integration tests for fork-aware transaction pool.
+
+use std::path::PathBuf;
+
+use derive_builder::Builder;
+use zombienet_configuration::{shared::types::Arg, types::ParaId};
+use zombienet_sdk::{LocalFileSystem, Network as ZNetwork, NetworkConfig};
+
+pub mod limits_30;
+pub mod old_pool;
+pub mod old_pool_small;
+pub mod single_collator;
+pub mod yap;
+
+const DEFAULT_BASE_DIR: &'static str = "/tmp/zn-spawner";
+const DEFAULT_RC_NODE_RPC_PORT: u16 = 9944;
+const DEFAULT_PC_NODE_RPC_PORT: u16 = 8844;
+
+/// TODO 1: each section has also a shared part that could be passed
+/// to each node: e.g. args that repeat through parachain nodes,
+/// and can be set sooner, at test inception.
+/// TODO 2: each network is mostly the same thing, with different args
+/// for parachains, and different numbers of parachains. The difference
+/// can be stored in dedicated functions that would setup the differences
+/// accordingly, part of a single `TxPoolPlayGroundNetwork` struct.
+
+#[derive(Default, Builder, Debug)]
+pub struct RelaychainConfig {
+	default_command: String,
+	chain: String,
+}
+
+impl RelaychainConfig {
+	pub fn new(default_command: String, chain: String) -> Self {
+		RelaychainConfig { default_command, chain }
+	}
+}
+
+#[derive(Default, Builder, Debug)]
+pub struct ParachainConfig {
+	default_command: String,
+	chain_spec_path: String,
+	cumulus_based: bool,
+	pub id: ParaId,
+}
+
+impl ParachainConfig {
+	pub fn new(
+		default_command: String,
+		chain_spec_path: String,
+		cumulus_based: bool,
+		id: ParaId,
+	) -> Self {
+		ParachainConfig { default_command, chain_spec_path, cumulus_based, id }
+	}
+}
+
+/// Wrapper over a substrate node managed by zombienet.
+#[derive(Debug)]
+pub struct Node {
```

Comment:
This should be named `NodeConfig` (or sth like this). From what I see it is used to provide params of single node.